### PR TITLE
Add a utility function to execute hooks serially

### PR DESCRIPTION
**Pull request by malept**
* [x] I have read the [contribution documentation](https://github.com/electron-userland/electron-packager/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [code of conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project follows, as appropriate.
* [x] The changes are appropriately documented (if applicable).
* [x] The changes have sufficient test coverage (if applicable).
* [x] The testsuite passes successfully on my local machine (if applicable).

**Summarize your changes:**

Adds `serialHooks(hooks)` (ported from Electron Forge), which allows a user to run hooks serially instead of in parallel (which was added in v10.0.0 due to `Promise.all`).

Closes #812.

### CODE REVIEW

File: test/hooks.js [hunk 1]
```diff
@@ -32,3 +34,49 @@ function createHookTest (hookName) {
...
+    return () => new Promise(resolve => { // eslint-disable-line promise/avoid-new
+      setTimeout(() => {
+        output += ` ${number}`
+        resolve()
+      }, 1000)
+    })
+  }
+  const testHooks = [
+    () => { output += ' 1'; return Promise.resolve() },
+    timeoutFunc(2),
```
Comment (MarshallOfSound) [inline test/hooks.js]: Can we use this in the above test as well

File: test/hooks.js [hunk 2]
```diff
@@ -32,3 +34,49 @@ function createHookTest (hookName) {
...
+    done => { setTimeout(() => { output += ' 10'; done() }, 1000) }
+  ]
+
+  return hooks.promisifyHooks(testHooks)
+    .then(() => t.not(output, '0 1 2 3 4 5 6 7 8 9 10', 'should not be in sequential order'))
+})
+
+test('serialHooks executes functions serially', t => {
+  let output = '0'
+  const timeoutFunc = number => {
```
Comment (MarshallOfSound) [inline test/hooks.js]: If this helper takes in the timeout as well we can remove more code duplication below for the hooks that instantly resolve

Comment (malept) [reply test/hooks.js]: :+1: https://github.com/electron-userland/electron-packager/pull/814/commits/9dcc68c1d0232382c3bd00d65602eca93d4213f5

**Comment by benlancaster** [comment]
This is great 👍

**Pull request merged**